go/cmd/server/logging/filelogstorage.go

```diff
@@ -0,0 +1,74 @@
+package logging
+
+import (
+	"errors"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/apache/arrow/go/v8/parquet"
+	"github.com/apache/arrow/go/v8/parquet/pqarrow"
+)
+
+type FileLogStorage struct {
+	// Feast project name
+	project string
+	path    string
+}
+
+func NewFileOfflineStore(project string, offlineStoreConfig map[string]interface{}) (*FileLogStorage, error) {
+	store := FileLogStorage{project: project}
+	var abs_path string
+	var err error
+	// TODO(kevjumba) remove this default catch.
+	if val, ok := offlineStoreConfig["path"]; !ok {
+		abs_path, err = filepath.Abs("log.parquet")
+	} else {
+		result, ok := val.(string)
+		if !ok {
+			return nil, errors.New("cannot convert offlinestore path to string")
+		}
+		abs_path, err = filepath.Abs(filepath.Join(result, "log.parquet"))
+	}
+	if err != nil {
+		return nil, err
+	}
+	store.path = abs_path
+	return &store, nil
+}
+
+func CreateOrOpenLogFile(absPath string) (*os.File, error) {
+	var _, err = os.Stat(absPath)
+
+	// create file if not exists
+	if os.IsNotExist(err) {
+		var file, err = os.Create(absPath)
+		if err != nil {
+			return nil, err
+		}
+		return file, nil
+	} else {
+		var file, err = os.OpenFile(absPath, os.O_RDWR, 0644)
```

Comment:
I think it's better to throw error here if file exists. Overwrite something or append can cause a flaky behaviour (especially in tests)

Reply:
Done.